### ISSUE TITLE
animationdatasinglefile.txt generated by the Nemesis engine causes male NPCs to not attack in melee.

### ISSUE THREAD

**Issue by ToasterStrudell** [ShikyoKira/Project-New-Reign---Nemesis-Main#369]
Hello, I found a bug in the Nemesis engine or rather in the file that it generates that made my game a little... let's just say "funky". You see, as the title suggests, the animationdatasinglefile.txt that the engine generates completely disables all humanoid male NPCs melee attack, whether it's one-handed, hand2hand, or two-handed, but keeps ranged, bash, block, and magic attacks, making all the Warrior, Tank, and Assassin class NPCs pretty much useless (the most noticeable one is the Assassin, as they tend to dual-wield and not able to block nor bash so they just stand there dodging and shuffling awkwardly with daggers on their hand).

The problem starts after installing Combat Gameplay Overhaul (CGO), more precisely after running Nemesis. I use a couple of mods that also requires Nemesis but none of them conflicts with CGO, or at least they claimed not to conflict with CGO as multiple people already used the same set of mod and come out fine, here's the list of Nemesis required mod that I used:
- The Ultimate Dodge Mod (TUDM)
- Crouch Sliding
- 360 Movement Behavior
- Combat Gameplay Overhaul
and none of them are conflicting with each other, other than they have their own default animationdatasinglefile.txt of their own, which shouldn't be a problem considering the file is going to be replaced the Nemesis generated one anyway. Now you might be thinking, "So the problem is CGO?", and I would confirm, **if** uninstalling CGO fixed the problem. You see, even after clean uninstalling CGO, cleaning my save, even to the point of reinstalling Nemesis itself the problem persists and it carries to "New Games" as well

Now, I won't issue the problem here if I had found a fix myself, as I have a more then adequate experience in troubleshooting, however, after scrounging the Nexus forums, CGO's mod page, Crouch Sliding mod page and the Nemesis github page, I managed to found:
- Several people are having the same issue (CGO's "posts" tab)
- This same issue is already open but with a different name and with a different mod sets (https://github.com/ShikyoKira/Project-New-Reign---Nemesis-Main/issues/180)

A "fix" has been found, which is to replace the auto generated animationdatasinglefile.txt with one of the defaults from one of the mod, which in my case, I used the CGO default one, and everything works dandy. However, using this fix will render a few mods inoperable (as it should, considering the file is for CGO and CGO only), for reference:

When using the CGO default:
Working mods: 360 Movement Behavior, TUDM (assumed), CGO.
Inoperable mods: Crouch Sliding (the slide stays in one place)
Issues: none so far

When using Nemesis Auto Generated file:
Working mods: TUDM, CGO, Crouch Sliding
Inoperable mods: 360 Movement Behavior
Issues: Male NPCs melee doesn't attack

Also here's a few small tidbits that I discover during my troubleshooting that I believe can help you pinpoint the problem:
- The auto generated file has a stark difference in file size when compared to either of the defaults (either TUDM or CGO) up to 100+ KB, with the Nemesis one at 1,1711 KB and the defaults at 1,850+ KB
- CGO and TUDM works fine with female NPCs and non melee male NPCs when using the auto generated file

I will upload 3 animationdatasinglefile.txt for you to use as a reference (TUDM, CGO, and Nemesis Auto Generated) and my recent PatchLog. I hope I explained the problem well as english is not my native tongue.

with all being said, I wish you luck on your endeavors.
Best regards,

The files:
[PatchLog.txt](https://github.com/ShikyoKira/Project-New-Reign---Nemesis-Main/files/5163297/PatchLog.txt)
[animationdatasinglefile (CGO Default).txt](https://github.com/ShikyoKira/Project-New-Reign---Nemesis-Main/files/5163302/animationdatasinglefile.CGO.Default.txt)
[animationdatasinglefile (Nemesis Generated).txt](https://github.com/ShikyoKira/Project-New-Reign---Nemesis-Main/files/5163311/animationdatasinglefile.Nemesis.Generated.txt)
[animationdatasinglefile (TUDM Default).txt](https://github.com/ShikyoKira/Project-New-Reign---Nemesis-Main/files/5163312/animationdatasinglefile.TUDM.Default.txt)

**Comment by ddmjr22** [comment]
Regarding the fix, when you say to replace the newly generated txt file, do you mean the one found in the nemesis mod itself? nemesis/meshes/(file) Because for me the file there is called "_nemesis__animationdatasinglefile.txt" Or do you mean to replace the one in the original mod after running nemesis? CGO/meshes/(file) or are these both incorrect? Because I've tried both of them and the issue still persist (testing on mid playthrough)

**Comment by riskirills66** [comment]
on my mods, the nemesis engine doesn't generate animationadata/animationdatasetsinglefile at all.

**Comment by saintgadreel** [comment]
I noticed while troubleshooting another issue with horse dismount and CBE that may be related to the strangely mobile animationdatasinglefile.txt. I found that Nemesis appears to be writing to the original files of installed mods instead of Overwrite.  It does not appear to just be animationdatasinglefile/animationdatasetsinglfile that don't appear in Nemesis output, as it seems to be updating .hkx files in the original mods' behavior folders.  I'm not sure if that's intended behavior or just a quirk of MO2, but I figured it might cause problems when re-running Nemsis a second time, or after making changes to installed mods or mod order.  

As far as observed behavior; the files were updated (filetime changed to match Nemesis run time) in CGO and CBE (unofficial patch vers if that matters) mod folders.  Files not present in CGO were updated in CBE, and CGO folder contents updated if it was overwriting CBE. The changes only affected contents of the various behaviors folders. 

So far I've copied any updated files by hand into the output folder, and once that is done (and the original mods are reinstalled to restore original files), subsequent runs on Nemesis update the output contents correctly in the Nemesis Output folder (unless anything new is introduced).

**Comment by Brucasco** [comment]
Almost 4 years and i have this same problem.....no one could fix this on Nemesis? :(

**Comment by Mirko132T** [comment]
> Almost 4 years and i have this same problem.....no one could fix this on Nemesis? :(

I couldnt find a fix either but I switched to Pandora and everything works normally.

**Comment by Brucasco** [comment]
I finally found a fix!. Someone (external from Nemesis's team as sadly they don't help to much about this issue) made a patch and uploaded it to Nexus:

https://www.nexusmods.com/skyrim/mods/114476?tab=description&jump_to_comment=144995199

The only catch is that you have to reinstall the patch (i recommend manually as with MO2 it didn't worked for me) everytime that you run Nemesis as that replaces the animationdatasinglefile.txt

I hope this can help to anyone that get's this issue!
